griptape/mixins/activity_mixin.py

```diff
@@ -3,16 +3,24 @@
 import inspect
 from typing import Callable, Optional
 
-import schema
 from attrs import Attribute, define, field
 from jinja2 import Template
-from schema import Literal, Schema
+from schema import Schema
 
 
 @define(slots=False)
 class ActivityMixin:
+    """Provides Tool Activity management functionality to Tools.
+
+    Attributes:
+        allowlist: List of Tool Activities to include in the Tool schema.
+        denylist: List of Tool Activities to remove from the Tool schema.
+        extra_schema_properties: Mapping of Activity name and extra properties to include in the activity's schema.
+    """
+
     allowlist: Optional[list[str]] = field(default=None, kw_only=True)
     denylist: Optional[list[str]] = field(default=None, kw_only=True)
+    extra_schema_properties: dict = field(default=None, kw_only=True)
```

Comment:
Not in love with this name

Reply:
is it worth updating to a wrapper `Activity` class that holds this information? schema, extra props, etc.

Reply:
also, does this contain descriptions about what the properties are? especially with the query params for search engines, they are basically just random characters that an LLM would not be able to figure out how to use.

Reply:
Yeah we definitely should have an `Activity` type vs cramming extra attributes into `Callable`. I created [an issue](https://github.com/griptape-ai/griptape/issues/1023).

> Also, does this contain descriptions about what the properties are?

Take a look at the updated PR description. I just realized I was missing an example of how this change gets used.